src/Features/Core/Portable/AddParameter/AbstractAddParameterCodeFixProvider.cs

```diff
@@ -113,127 +135,378 @@ internal abstract class AbstractAddParameterCodeFixProvider<
 
-            var newRoot = editor.GetChangedRoot();
-            var newDocument = methodDocument.WithSyntaxRoot(newRoot);
 
-            return newDocument;
+                    if (method.MethodKind == MethodKind.ReducedExtension)
+                    {
+                        insertionIndex++;
+                    }
+
+                    AddParameter(
+                        syntaxFacts, editor, methodNode, argument,
+                        insertionIndex, parameterDeclaration, cancellationToken);
+                }
+
+                var newRoot = editor.GetChangedRoot();
+                solution = solution.WithDocumentSyntaxRoot(document.Id, newRoot);
+            }
+
+            return solution;
         }
 
-        private async Task<(IParameterSymbol, bool isNamedArgument)> CreateParameterSymbolAsync(
-            Document invocationDocument,
-            IMethodSymbol method,
-            TArgumentSyntax argument,
-            CancellationToken cancellationToken)
+        private static async Task<(ITypeSymbol, RefKind)> GetArgumentTypeAndRefKindAsync(Document invocationDocument, TArgumentSyntax argument, CancellationToken cancellationToken)
         {
             var syntaxFacts = invocationDocument.GetLanguageService<ISyntaxFactsService>();
-            var semanticFacts = invocationDocument.GetLanguageService<ISemanticFactsService>();
-            var argumentName = syntaxFacts.GetNameForArgument(argument);
-            var expression = syntaxFacts.GetExpressionOfArgument(argument);
-
             var semanticModel = await invocationDocument.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
-            var parameterType = semanticModel.GetTypeInfo(expression).Type ?? semanticModel.Compilation.ObjectType;
+            var argumentExpression = syntaxFacts.GetExpressionOfArgument(argument);
+            var argumentType = semanticModel.GetTypeInfo(argumentExpression).Type ?? semanticModel.Compilation.ObjectType;
+            var refKind = syntaxFacts.GetRefKindOfArgument(argument);
+            return (argumentType, refKind);
+        }
+
+        private static async Task<ImmutableArray<IMethodSymbol>> FindMethodDeclarationReferences(
+            Document invocationDocument, IMethodSymbol method, CancellationToken cancellationToken)
+        {
+            var progress = new StreamingProgressCollector(StreamingFindReferencesProgress.Instance);
+
+            await SymbolFinder.FindReferencesAsync(
+                symbolAndProjectId: SymbolAndProjectId.Create(method, invocationDocument.Project.Id),
+                solution: invocationDocument.Project.Solution,
+                documents: null,
+                progress: progress,
+                cancellationToken: cancellationToken).ConfigureAwait(false);
+            var referencedSymbols = progress.GetReferencedSymbols();
+            return referencedSymbols.Select(referencedSymbol => referencedSymbol.Definition).OfType<IMethodSymbol>().ToImmutableArray();
```

Comment:
so if finds all references, and drop everything except definitions? any reason not keeping references if it already paid all cost? you can add parameter and give in default.

also, shouldn't this do some distinct to unique IMethodSymbols?

Reply:
I think this should do de-dup the fixer part seems not deduping as well causing same definition to show up twice which you already rewrote.

Reply:
actually, it might work because in fixer, you doesn't get updated root but use same original root to write each method, so it ends up do same thing multiple time but end result is always same regardless.

Reply:
> also, shouldn't this do some distinct to unique IMethodSymbols?

You are right. Done in a98dabcdff0d261b190d50bd3a47399d3c2bae7d.